pkg/tsdb/testdata/scenarios.go

```diff
@@ -517,9 +521,33 @@ func getRandomWalk(query *tsdb.Query, tsdbQuery *tsdb.TsdbQuery) *tsdb.QueryResu
 
 	queryRes := tsdb.NewQueryResult()
 	queryRes.Series = append(queryRes.Series, series)
+	attachLabels(query, queryRes)
 	return queryRes
 }
 
+func attachLabels(query *tsdb.Query, queryRes *tsdb.QueryResult) {
+	labelText := query.Model.Get("labels").MustString("")
+	if labelText == "" {
+		return
+	}
+
+	re := regexp.MustCompile(`\b(\w+)(!?=~?)"([^"\n]*?)"`)
```

Comment:
Can concat the strings of the regex to make it more readable by adding comments, see https://stackoverflow.com/a/24589824/107156 . 

Also the regex should be a constant outside of the function (to only compile once).